### PR TITLE
VC slashing part 5 - vc slashing sse

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Validator Slashing Part 5:
Handle `attester_slashing` and `proposer_slashing` events and subscribe to them only when the `Xshut-down-when-validator-slashed-enabled` option is set to true
## Fixed Issue(s)

#7707 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapter.java [hunk 1]
```diff
@@ -128,7 +133,16 @@ public void onPrimaryNodeBackReady() {
...
 
   private BackgroundEventSource createEventSource(final RemoteValidatorApiChannel beaconNodeApi) {
-    final HttpUrl eventSourceUrl = createHeadEventSourceUrl(beaconNodeApi.getEndpoint());
+
+    final List<EventType> eventTypes = new ArrayList<>();
+    eventTypes.add(EventType.head);
+    if (shutdownWhenValidatorSlashedEnabled) {
+      eventTypes.add(EventType.attester_slashing);
+      eventTypes.add(EventType.proposer_slashing);
+    }
```
Comment (rolfyone) [inline validator/remote/src/main/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapter.java:144]: ideally test case here to so we don't regress...

Comment (mehdi-aouadi) [reply validator/remote/src/main/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapter.java:144]: done

File: validator/remote/src/test/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  @ParameterizedTest(name = "subscribe_to_slashing_events: {0}")
+  @ValueSource(strings = {"true", "false"})
+  public void shouldSubscribeToSlashingEvents(final boolean shutdownWhenValidatorSlashedEnabled) {
+    final EventSourceBeaconChainEventAdapter eventSourceBeaconChainEventAdapter =
+        initEventSourceBeaconChainEventAdapter(shutdownWhenValidatorSlashedEnabled);
+    eventSourceBeaconChainEventAdapter.createEventSource(beaconApiMock);
+    verifyEventSourceSubscriptionUrl(httpUrlMock, shutdownWhenValidatorSlashedEnabled);
+  }
+
+  @NotNull
```
Comment (rolfyone) [inline validator/remote/src/test/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapterTest.java]: i'd kill that annotation...

**Review by rolfyone — APPROVED** [review]
one nit, looks good!

File: validator/remote/src/test/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapterTest.java (hunk 2 above)
Comment (mehdi-aouadi) [reply validator/remote/src/test/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapterTest.java]: done

**Pull request merged**